sorl/thumbnail/templatetags/thumbnail.py

```diff
@@ -135,7 +135,11 @@ def _render(self, context):
             else:
                 options[key] = value
 
-        thumbnail = get_thumbnail(file_, geometry, **options)
```

Comment:
I'd suggest the following:

```python
thumbnail = None
if file_:
    thumbnail = get_thumbnail(file_, geometry, **options)

if not thumbnail:
    if self.nodelist_empty:
        return self.nodelist_empty.render(context)
    else:
        return ''
```

Currently this PR is providing a dummy images where the ImageField is just an empty string - I'd expect these to return nothing or render the empty block.

I think we only want dummy images as a fallback for images which are missing but should exist - which is what `get_thumbnail` does.

Reply:
That causes current tests to fail. I like the behaviour you're suggesting, though, @tomkins.

Reply:
The tests expect `{% thumbnail "" "x400" as im %}...` to render a dummy image, but the change suggested in your comment causes the code to render an empty string instead. This may be a good direction to go in, but this is definitely a breaking change. I prefer making a small change that breaks no existing tests for now.

For reference, here are the test failures that the change suggested in your comment causes:

```
============================================================================================================ FAILURES ============================================================================================================
flimm/sorl-thumbnail/tests/thumbnail_tests/test_engines.py:412: AssertionError: '' != '<img src="http://dummyimage.com/600x400" [100 chars]5x">'
flimm/sorl-thumbnail/tests/thumbnail_tests/test_engines.py:399: AssertionError: '<p>NOT</p>' != '<img src="http://dummyimage.com/300x200" width="300" height="200"><p>NOT</p>'
flimm/sorl-thumbnail/tests/thumbnail_tests/test_templatetags.py:24: AssertionError: '<img style="margin:auto" width="200" height="100">' != '<img style="margin:0px 0px 0px 0px" width="200" height="100">'
flimm/sorl-thumbnail/tests/thumbnail_tests/test_templatetags.py:120: AssertionError: '<img style="margin:auto" width="20" height="20">' != '<img style="margin:0px 0px 0px 0px" width="20" height="20">'
flimm/sorl-thumbnail/tests/thumbnail_tests/test_templatetags.py:143: AssertionError: b'<img style="margin:auto" width="30" height="30">' != b'<p>empty</p>'
flimm/sorl-thumbnail/tests/thumbnail_tests/test_templatetags.py:161: AssertionError: '<img style="margin:auto" width="200" height="100">' != '<img style="margin:0px 0px 0px 0px" width="200" height="100">'
========================================================================================= 6 failed, 62 passed, 6 skipped in 9.25 seconds =========================================================================================
ERROR: InvocationError: 'flimm/sorl-thumbnail/.tox/py36-django20-imagemagick/bin/py.test -rw --cov-config setup.cfg --cov sorl --tb=line'
____________________________________________________________________________________________________________ summary _____________________________________________________________________________________________________________
ERROR:   py36-django20-imagemagick: commands failed
```